dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamUtils.java

```diff
@@ -158,62 +139,154 @@ private static Stream<String> parseUidString( String input )
     }
 
     /**
-     * Parse request parameter to filter tracked entity attributes using
-     * identifier, operator and values. Refer to
+     * Parse request parameter to filter tracked entity attributes using UID,
+     * operator and values. Refer to
      * {@link #parseQueryItem(String, CheckedFunction)} for details on the
      * expected item format.
      *
-     * @param items query item strings each composed of identifier, operator and
+     * @param filterItem query item string composed of identifier, operator and
      *        value
      * @param attributes tracked entity attribute map from identifiers to
      *        attributes
-     * @return query items each of a tracked entity attribute with attached
-     *         query filters
+     * @return query item of tracked entity attribute with attached query
+     *         filters
      */
-    public static List<QueryItem> parseAttributeQueryItems( Set<String> items,
+
+    public static List<QueryItem> parseAttributeQueryItems( String filterItem,
         Map<String, TrackedEntityAttribute> attributes )
         throws BadRequestException
     {
+        if ( StringUtils.isEmpty( filterItem ) )
+        {
+            return List.of();
```

Comment:
This is causing an issue in the exporter endpoint as this collection shouldn't be immutable in this context, in the store `QueryItem`s can be added to it to make the order work.
When trying to order on an attribute, the server returns a 500 error.

In master PR we found the issue but it is going to be solved in the controller refactoring, here it needs to be changed.